providers/universal-provider/src/providers/eip155.ts

```diff
@@ -168,6 +170,34 @@ class Eip155Provider implements IProvider {
   private isChainApproved(chainId: number): boolean {
     return this.namespace.chains.includes(`${this.name}:${chainId}`);
   }
+
+  private async getCapabilities(args: RequestParams) {
+    // if capabilities are stored in the session, return them, else send the request to the wallet
+    const address = args.request?.params?.[0];
+    if (!address) throw new Error("Missing address parameter in `wallet_getCapabilities` request");
+    const session = this.client.session.get(args.topic);
+    const sessionCapabilities = session?.sessionProperties?.capabilities || {};
+    if (sessionCapabilities?.[address]) {
+      return sessionCapabilities?.[address];
+    }
+    // intentionally omit catching errors/rejection during `request` to allow the error to bubble up
+    const capabilities = await this.client.request(args as EngineTypes.RequestParams);
+    try {
+      // update the session with the capabilities so they can be retrieved later
+      await this.client.session.update(args.topic, {
+        sessionProperties: {
+          ...(session.sessionProperties || {}),
+          capabilities: {
+            ...(sessionCapabilities || {}),
+            [address]: capabilities,
+          } as any, // by spec sessionProperties should be <string, string> but here are used as objects?
```

Comment:
what do we do about this? The `spec` is the EIP-5792 spec. I think we do something else maybe in our CAIP-25 spec? So maybe we just cover the "right" think in our wallet docs / wallet SDK?

We probably need to expand our namespace builder for this usecase

Reply:
according to `caip-25` spec the sessionProperties (scopeProperties) is `<string, string>`